### ISSUE TITLE
naming conventions: rotate-x => rotate-x!

### ISSUE THREAD

**Issue by kristianlm** [AlexCharlton/gl-math#3]
Just a rename proposal:

```
rotate-x => rotate-x!
rotate-y => rotate-y!
rotate-z => rotate-z!
```

Since the procedures are modifying the matrix.

**Comment by AlexCharlton** [comment]
Yeah, that would make sense. I think the full list of matrix modification functions looks like this:
```
translate
rotate-x
rotate-y
rotate-z
rotate-axis-angle
rotate-quaternion
rotate-ypr
scale-3d
scale
flip-y
flip-x
```
If you feel like submitting a PR, please do (although there's almost certainly several other dependencies that would need to be changed. Otherwise, I haven't been too happy with this API and will probably revisit it at some point. The names with the bangs would probably be what ends up getting used, either way.
